lib/utils/update-check.js

```diff
@@ -25,18 +26,23 @@ module.exports = function updateCheck() {
                     `Run ${chalk.cyan('`npm install -g ghost-cli@latest`')} to upgrade.\n`
                 ));
 
-                inquirer.prompt({
-                    type: 'confirm',
-                    message: 'Continue without upgrading?',
-                    default: false,
-                    name: 'yes'
-                }).then((answers) => {
-                    if (answers.yes) {
-                        return resolve();
-                    }
+                // Yargs hasn't run yet
+                if (includes(process.argv, '--no-prompt')) {
+                    reject('Requested no prompting. Exiting...');
```

Comment:
Might be cleaner just to do:

```js
if (includes(process.argv, '--no-prompt')) {
  reject('Requested no prompting. Exiting...');
  return;
}

inquirer.prompt() // ...
```

Reply:
This should not reject? Rejecting here means any call to any command with `--no-prompt` will always fail.

Reply:
No - this block of code only gets executed if the version is out of date.

So if your version of ghost-cli is up to date then you'll be fine. If it's out-of-date and you pass the `--no-prompt` flag then it rejects.

Reply:
See https://github.com/TryGhost/Ghost-CLI/issues/565#issuecomment-350779915. This _only_ makes sense when running the `ghost upgrade` command as far as I can see.

Reply:
I was thinking of moving the update check to just before yargs runs in [lib/bootstrap](/tryghost/ghost-cli/blob/master/lib/bootstrap.js#L145) based on [#565 (comment)](/tryghost/ghost-cli/issues/565#issuecomment-350769339). It would only resolve / continue if the specific command doesn't require ghost to be up to date (i.e. `CommandClass.versionSpecific = false`)